programs/stake/src/stake_instruction.rs

```diff
@@ -191,7 +191,36 @@ pub fn split(
         system_instruction::create_account(
             stake_pubkey,
             split_stake_pubkey,
-            0, // creates an ephemeral, uninitialized Stake
+            0, // may create an ephemeral, uninitialized Stake
+            std::mem::size_of::<StakeState>() as u64,
+            &id(),
+        ),
+        {
+            let account_metas = vec![
+                AccountMeta::new(*stake_pubkey, false),
+                AccountMeta::new(*split_stake_pubkey, false),
+            ]
+            .with_signer(authorized_pubkey);
+            Instruction::new(id(), &StakeInstruction::Split(lamports), account_metas)
+        },
+    ]
+}
+
+pub fn split_with_seed(
+    stake_pubkey: &Pubkey,
+    authorized_pubkey: &Pubkey,
+    lamports: u64,
+    split_stake_pubkey: &Pubkey, // derived using create_address_with_seed()
+    base: &Pubkey,               // base
+    seed: &str,                  // seed
+) -> Vec<Instruction> {
+    vec![
+        system_instruction::create_account_with_seed(
+            stake_pubkey,
```

Comment:
If the `authorized_pubkey` is the assumed fee payer, how about using it as the `from` keypair? Seems like that would elide the need for special case handling of zero lamport account creations. It could be assumed that `from` is always the fee payer and needs to be signed for fees even if it's not sending any lamports.

Reply:
authorized_pubkey is not necessarily the fee payer, we support arbitrary authorities in stake

Reply:
Ah got it, thanks